datajoint_utilities/typing/__init__.py

```diff
@@ -0,0 +1,180 @@
+from datajoint_utilities.generic.pkgutil import detect_min_python_version
+
+detect_min_python_version(3, 10)
+
+import datetime
+import inspect
+import types
+from pathlib import Path
+from typing import (
+    Any,
+    ItemsView,
+    Literal,
+    Mapping,
+    MutableMapping,
+    Protocol,
+    Sequence,
+    Sized,
+    TypeAlias,
+    TypedDict,
+    TypeGuard,
+    TypeVar,
+    cast,
+    runtime_checkable,
+)
+from uuid import UUID
+
+import datajoint as dj
+from numpy import bool_, complex_, float_, int_, ndarray, str_
+from numpy.typing import NDArray
+
+# Misc. typing -------------------------------------------------------------------------
+T = TypeVar("T")
+DictObj: TypeAlias = dict[str, object]
```

Comment:
- They help with readability (some types are comnbinations of aliases, which can lead to very long type unions)
- Consistency in type unions across several function arguments and return types
- Centralize types in a module instead of several import lines across multiple files